src/asl.ts

```diff
@@ -1446,6 +1525,100 @@ export class ASL {
     ) as any;
   }
 
+  /**
+   * Recursively evaluate a single expression, building a single {@link ASLGraph.NodeResults} object.
+   * All SubStates generated during evaluation will be merged into into a {@link ASLGraph.OutputSubState} along with the output
+   * of the handler callback.
+   *
+   * @param expr - Expression to evaluate.
+   * @param contextNode - Optional node to associate with the output state. This node may be used to name the resulting state.
+   *                      Otherwise expr is used.
+   * @param handler - A handler callback which received the {@link ASLGraph.Output} resolved from the expression.
+   *                  This output will represent the constant or variable representing the output of the expression.
+   *                  An `addState` callback is also provided to inject additional states into the graph.
+   *                  The state will be joined (@see ASLGraph.joinSubStates ) with the previous and next states in the order received.
+   */
+  public evalExprToJsonPath(
+    expr: Expr,
+    handler: EvalExprHandler<ASLGraph.JsonPath>
+  ): ASLGraph.NodeResults;
+  public evalExprToJsonPath(
+    expr: Expr,
+    contextNode: FunctionlessNode,
+    handler: EvalExprHandler<ASLGraph.JsonPath>
+  ): ASLGraph.NodeResults;
+  public evalExprToJsonPath(
+    expr: Expr,
+    nodeOrHandler: FunctionlessNode | EvalExprHandler<ASLGraph.JsonPath>,
+    maybeHandler?: EvalExprHandler<ASLGraph.JsonPath>
+  ): ASLGraph.NodeResults {
+    const [node, handler] = isNode(nodeOrHandler)
+      ? [nodeOrHandler, maybeHandler!]
+      : [expr, nodeOrHandler];
+
+    return this.evalExpr(expr, node, (output, context) => {
+      const normalizedState = this.normalizeOutputToJsonPath(output);
+
+      if (ASLGraph.isStateOrSubState(normalizedState)) {
+        context.addState(normalizedState);
+      }
+
+      const newOutput = ASLGraph.isStateOrSubState(normalizedState)
+        ? normalizedState.output
+        : normalizedState;
+
+      return handler(newOutput, context);
+    });
+  }
+
+  /**
+   * Recursively evaluate a single expression, building a single {@link ASLGraph.NodeResults} object.
+   * All SubStates generated during evaluation will be merged into into a {@link ASLGraph.OutputSubState} along with the output
```

Comment:
nit:

> into into

Reply:
nice, thats been there for a least a month in the one it was copied from.